dev/build.example.testcontainers_fat/fat/src/com/ibm/ws/testcontainers/example/DockerfileTest.java

```diff
@@ -0,0 +1,103 @@
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.testcontainers.example;
+
+import java.time.Duration;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.runner.RunWith;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy;
+
+import com.ibm.websphere.simplicity.ShrinkHelper;
+
+import componenttest.annotation.Server;
+import componenttest.annotation.TestServlet;
+import componenttest.containers.SimpleLogConsumer;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+import web.generic.ContainersTestServlet;
+
+/**
+ * Example test class showing how to setup a testcontainer that uses a custom dockerfile.
+ */
+@RunWith(FATRunner.class)
+public class DockerfileTest {
+
+    public static final String APP_NAME = "containerApp";
+
+    @Server("build.example.testcontainers")
+    @TestServlet(servlet = ContainersTestServlet.class, contextRoot = APP_NAME)
+    public static LibertyServer server;
+
+    public static final String POSTGRES_DB = "test";
+    public static final String POSTGRES_USER = "test";
+    public static final String POSTGRES_PASSWORD = "test";
+
+    /**
+     * There are times where we might want to extend a base docker image for our
+     * own testing needs. For example, using a docker image that already uses a startup script.
+     * It is possible to provide testcontainers with a DockerFile and build a new image at runtime.
+     *
+     * <pre>
+     * public static GenericContainer<?> container = new GenericContainer<>(
+     *          new ImageFromDockerfile()
+     *          .withDockerfile(Paths.get("lib/LibertyFATTestFiles/postgres/Dockerfile")))
+     * </pre>
+     *
+     * Doing this will result in a warning being logged:
+     * W WARNING: Cannot use private registry for programmatically built image testcontainers/[image-sha]:latest.
+     * Consider using a pre-built image instead.
+     *
+     * This warning is thrown to alert the developer that we CANNOT use our Artifactory image cache.
+     * This could result in hitting our docker pull limits and introduce intermittent failures.
+     *
+     * Instead, it is best practice to build and push your docker image to docker hub, and reference that instead.
+     * You can still keep the Dockerfile and any related files in source control under publish/files.
+     *
+     * In this case I pushed my custom image to my personal DockerHub repo under kyleaure/postgres-test-table:1.0
+     * You will notice that everything else is configured the same as in the regular ContainersTest test class.
+     */
+    @ClassRule
+    public static GenericContainer<?> container = new GenericContainer<>("kyleaure/postgres-test-table:1.0")
```

Comment:
If `kyleaure/postgres-test-table:1.0` is a valid DockerHub image, any concerns that your DockerHub limits will get hit due to people using this test value?

"kyleaure/postgres-test-table:1.0" -> POSTGRES_DOCKERHUB_IMAGE ?

Reply:
There are limits on how often a system can pull images from dockerhub. 
But, dockerhub doesn't ding the person who created the image for the number of pulls it is getting.

Reply:
Got it, thanks for helping me understand that.